### PR TITLE
Add ProjectRequest cleanup

### PR DESCRIPTION
**Pull request by jpeimer**
##### Short description:
Add ProjectRequest cleanup

## Summary by CodeRabbit

- **New Features**
	- Introduced a `clean_up` method in the Project Request functionality, allowing for enhanced resource management with customizable wait and timeout options.

### CODE REVIEW

**Comment by rh-bot-1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install
 * `/retest all`: Retest all

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Review by dbasunag — APPROVED** [review]

File: ocp_resources/project_request.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 from typing import Any, Optional
 from ocp_resources.resource import Resource
+from ocp_resources.project_project_openshift_io import Project
```
Comment (rnetser) [inline ocp_resources/project_request.py:5]: what about Project in project_config_openshift_io?

Comment (jpeimer) [reply ocp_resources/project_request.py:5]: Should we use it instead? Before the class-generator PR, cleanup called the `Project` from `project_project_openshift_io`

But I do see that even with this PR, unprivileged user can't cleanup the project...

Comment (dbasunag) [reply ocp_resources/project_request.py:5]: I think this is only relevant to project.openshift.io, based on what I see 
```
[cloud-user@ocp-psi-executor-xl ~]$ oc explain ProjectRequest
GROUP:      project.openshift.io
KIND:       ProjectRequest
VERSION:    v1
```

**Review by rnetser — COMMENTED** [review]
i see that `client_wait_deleted` is also missing.

is this verified?

**Review by dbasunag — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Pull request merged**